specifyweb/specify/migrations/0002_geo.py

```diff
@@ -33,63 +32,75 @@ class Migration(migrations.Migration):
+                ('taxontreedef', models.ForeignKey(db_column='TaxonTreeDefID', on_delete=protect_with_blockers, related_name='cotypes', to='specify.taxontreedef')),
             ],
             options={
                 'db_table': 'collectionobjecttype',
                 'ordering': (),
-                'unique_together': (('collection', 'isdefault'),),
+            },
+        ),
+        migrations.CreateModel(
+            name='CollectionObjectGroupType',
+            fields=[
+                ('id', models.AutoField(db_column='COGTypeID', primary_key=True, serialize=False)),
+                ('name', models.CharField(db_column='Name', max_length=255, null=False)),
+                ('cogtype', models.CharField(blank=True, db_column='COGType', max_length=255, null=False)),
+                ('version', models.IntegerField(blank=True, db_column='Version', default=0, null=True)),
+                ('timestampcreated', models.DateTimeField(db_column='TimestampCreated', default=django.utils.timezone.now)),
+                ('timestampmodified', models.DateTimeField(blank=True, db_column='TimestampModified', default=django.utils.timezone.now, null=True)),
+                ('collection', models.ForeignKey(db_column='CollectionID', on_delete=protect_with_blockers, related_name='cogtypes', to='specify.collection')),
+                ('createdbyagent', models.ForeignKey(db_column='CreatedByAgentID', null=True, on_delete=protect_with_blockers, related_name='+', to='specify.agent')),
+                ('modifiedbyagent', models.ForeignKey(db_column='ModifiedByAgentID', null=True, on_delete=protect_with_blockers, related_name='+', to='specify.agent')),
+            ],
+            options={
+                'db_table': 'collectionobjectgrouptype',
+                'ordering': (),
             },
         ),
         migrations.AddField(
             model_name='collectionobject',
-            name='cotype',
-            field=models.ForeignKey(db_column='COTypeID', null=True, on_delete=models.SET_NULL, related_name='collectionobjects', to='specify.collectionobjecttype'),
+            name='collectionobjecttype',
+            field=models.ForeignKey(db_column='CollectionObjectTypeID', null=True, on_delete=models.SET_NULL, related_name='collectionobjects', to='specify.collectionobjecttype'),
         ),
         migrations.AddField(
-            model_name='collectionobject',
-            name='hasreferencecatalognumber',
-            field=models.BooleanField(blank=True, db_column='HasReferenceCatalogNumber', default=False, null=True),
+            model_name='collection',
+            name='collectionobjecttype',
+            field=models.ForeignKey(db_column='CollectionObjectTypeID', null=True, on_delete=protect_with_blockers, related_name='collections', to='specify.collectionobjecttype'),
```

Comment:
Is this the right `on_delete` functionality we want for `Collection.collectionobjecttype`? 

When you try and delete a CollectionObjectType, if there are any referencing Collections, block the deletion?